.circleci/config.yml

```diff
@@ -287,7 +282,37 @@ jobs:
       - checkout
       - *setup_remote_docker
       - *install_docker_client
-      - *docker_build
+      - run:
+          name: Build Docker image
+          command: |
+            ~/bin/docker build .
+
+  docker_push_tag:
+    <<: *container_config
+    steps:
+      - checkout
+      - *setup_remote_docker
+      - *install_docker_client
+      - run:
+          name: Build and push Docker image to DockerHub
+          command: |
+            ~/bin/docker login -u $DOCKER_USER -p $DOCKER_PASS
+            ~/bin/docker build -t ${DOCKERHUB_REPO:?}:${CIRCLE_TAG:?} .
+            ~/bin/docker tag ${DOCKERHUB_REPO:?}:latest ${DOCKERHUB_REPO:?}:${CIRCLE_TAG:?}
+            ~/bin/docker push ${DOCKERHUB_REPO:?}:${CIRCLE_TAG:?}
+
+  docker_push_dev:
+    <<: *container_config
+    steps:
+      - checkout
+      - *setup_remote_docker
+      - *install_docker_client
+      - run:
+          name: Build and push Docker image to DockerHub
+          command: |
+            ~/bin/docker login -u $DOCKER_USER -p $DOCKER_PASS
+            ~/bin/docker build -t ${DOCKERHUB_REPO:?}:dev .
+            ~/bin/docker push ${DOCKERHUB_REPO:?}:dev
```

Comment:
Same comment as for other push - if possible print before and after push so to see change in remote state.

Reply:
@lucafavatella Why do we need this? To see if the push worked? Docker push prints out the layer hashes, and reports if a layer already exists on the remote.

Reply:
@amir-hadi  Can you please quote lines that make this explicit? Looking at the output of the sample build that @dincho posted I could not find it (probably me not being familiar with docker push).

Reply:
@lucafavatella I just checked the logs and I coundn't find it. It must be a problem how docker outputs the information (it has a fancy progress bar in newer versions etc).

This is how docker push looks like on my server:
```The push refers to a repository [docker.io/ahadi/znc-docker]
d7c212d822a6: Layer already exists
d05104117d00: Layer already exists
2edfb6fb8973: Layer already exists
6adf46d34d91: Layer already exists
d2c12f0055eb: Layer already exists
607b9bebb156: Layer already exists
00f3d8510106: Layer already exists
e4b57986d764: Layer already exists
a3a69de6a181: Layer already exists
f7127c748181: Layer already exists
48e4f637e500: Layer already exists
08d7a688e98d: Layer already exists
eccdee480abd: Layer already exists
102dfad69cf7: Layer already exists
480a87c93e34: Layer already exists
f8ee36ed8c16: Layer already exists
2f2b8a63dcc6: Layer already exists
8c92a215240e: Layer already exists
725b4ef7ffce: Layer already exists
41ef8cc0bccb: Layer already exists
100396c46221: Layer already exists
7b4b54c74241: Layer already exists
d17d48b2382a: Layer already exists
latest-36efa85: digest: sha256:6cc0e30907c0d5a85f6e826937d5e2fa21122cf14d0f366f49e1381aef228622e0 size: 5148```